proparse/src/main/java/org/prorefactor/refactor/RefactorSession.java

```diff
@@ -74,19 +74,35 @@ public Schema getSchema() {
   }
 
   /**
-   * Get the listing file name, makes sure the directory exists.
-   */
-  public String getListingFileName() {
-    return "listingfile.txt";
-  }
-
-  /**
    * Returns the Settings for the currently loaded project
    */
   public IProparseSettings getProparseSettings() {
     return proparseSettings;
   }
 
+  public File findFile3(String fileName) {
+    
+    // If we have an absolute path-filename, we don't search the path.
+    // If we have a relative (starts with dot) path-filename, ditto.
+    int len = fileName.length();
+    
+    // Windows drive letter, ex: "C:"
+ // Relative path, "./" or "../"
+    if ((len > 0 && (fileName.charAt(0) == '/' || fileName.charAt(0) == '\\')) || (len > 1 && fileName.charAt(1) == ':')
+        || (len > 1 && fileName.charAt(0) == '.')) {
+      if (new File(fileName).exists())
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1066)